featurebyte/session/spark.py

```diff
@@ -269,26 +270,33 @@ def _process_batch_data(self, data: pd.DataFrame, schema: Schema) -> pd.DataFram
         pd.DataFrame
             Processed data
         """
+        if data.empty:
+            return data
+
         for i, column in enumerate(schema.names):
+            current_type = schema.field(i).type
             # Convert decimal columns to float
-            if schema.field(i).type == pa.float64() and not is_float_dtype(data[column]):
+            if current_type == pa.float64() and not is_float_dtype(data[column]):
                 data[column] = data[column].astype(float)
-            elif isinstance(schema.field(i).type, pa.TimestampType) and not is_datetime64_dtype(
+            elif isinstance(current_type, pa.TimestampType) and not is_datetime64_dtype(
                 data[column]
             ):
                 data[column] = pd.to_datetime(data[column])
+            elif current_type == pa.list_(pa.float64()):
```

Comment:
Would it make sense to make `pa.list_(pa.float64())` a constant that is shared between here and above? I assume they have to match exactly.

Reply:
yeap makes sense, will update!